src/backend/core/models.py

```diff
@@ -471,6 +545,28 @@ def get_abilities(self, user):
         }
 
 
+class TeamWebhook(BaseModel):
+    """Webhooks fired on changes in teams."""
+
+    team = models.ForeignKey(Team, related_name="webhooks", on_delete=models.CASCADE)
+    url = models.URLField(_("url"))
+    # TODO: use encrypted field https://pypi.org/project/django-encrypted-model-fields/
+    secret = models.CharField(_("secret"), max_length=255, null=True, blank=True)
+    status = models.CharField(
+        max_length=10,
+        default=WebhookStatusChoices.PENDING,
+        choices=WebhookStatusChoices.choices,
+    )
+
+    class Meta:
+        db_table = "people_team_webhook"
+        verbose_name = _("Team webhook")
+        verbose_name_plural = _("Team webhooks")
+
+    def __str__(self):
+        return f"Webhook to {self.url} for {self.team}"
```

Comment:
I didn't expect that ! You plan on having a webhook per team while I thought we had a webhook per application.

I see the value of having a status per team.

But if we plan on starting with Outline :
- there's no point in having a secret field, at least for Outline. As only administrators can change groups composition, I don't think we'll be using a different secret for each team (or we'll have to generate them programmatically and we'll have to check it is possible).
- for Outline again, instead of storing an URL, we should store UUIDs directly (to call API and infer URLs from it)

Reply:
Tough questions. Let''s discuss IRL but some elements of answers:
You can have several webhooks per team if you have several remote applications to sync.
Outline has no SCIM provider yet so the url/secret provisionning can be added there... we can do something simple for a v1...
Before going to production we will need to obfuscate the secret so that users only see it once in the UI (when they set it) and never again...